mbf_mesh_nav/include/mbf_mesh_nav/mesh_navigation_server.h

```diff
@@ -162,39 +161,28 @@ class MeshNavigationServer : public mbf_abstract_nav::AbstractNavigationServer
 
   /**
    * @brief Callback method for the check_pose_cost service
-   * @param request Request object, see the mbf_msgs/CheckPose service
+   * @param request Request object, see the mbf_msgs/srv/CheckPose service
    * definition file.
-   * @param response Response object, see the mbf_msgs/CheckPose service
+   * @param response Response object, see the mbf_msgs/srv/CheckPose service
    * definition file.
-   * @return true, if the service completed successfully, false otherwise
    */
-  bool callServiceCheckPoseCost(mbf_msgs::CheckPose::Request& request, mbf_msgs::CheckPose::Response& response);
+  void callServiceCheckPoseCost(std::shared_ptr<rmw_request_id_t> request_header, std::shared_ptr<mbf_msgs::srv::CheckPose::Request> request, std::shared_ptr<mbf_msgs::srv::CheckPose::Response> response);
 
   /**
    * @brief Callback method for the check_path_cost service
-   * @param request Request object, see the mbf_msgs/CheckPath service
+   * @param request Request object, see the mbf_msgs/srv/CheckPath service
    * definition file.
-   * @param response Response object, see the mbf_msgs/CheckPath service
+   * @param response Response object, see the mbf_msgs/srv/CheckPath service
    * definition file.
-   * @return true, if the service completed successfully, false otherwise
    */
-  bool callServiceCheckPathCost(mbf_msgs::CheckPath::Request& request, mbf_msgs::CheckPath::Response& response);
+  void callServiceCheckPathCost(std::shared_ptr<rmw_request_id_t> request_header, std::shared_ptr<mbf_msgs::srv::CheckPath::Request> request, std::shared_ptr<mbf_msgs::srv::CheckPath::Response> response);
 
   /**
    * @brief Callback method for the make_plan service
    * @param request Empty request object.
    * @param response Empty response object.
-   * @return true, if the service completed successfully, false otherwise
-   */
-  bool callServiceClearMesh(std_srvs::Empty::Request& request, std_srvs::Empty::Response& response);
-
-  /**
-   * @brief Reconfiguration method called by dynamic reconfigure.
-   * @param config Configuration parameters. See the MoveBaseFlexConfig
-   * definition.
-   * @param level bit mask, which parameters are set.
    */
-  void reconfigure(mbf_mesh_nav::MoveBaseFlexConfig& config, uint32_t level);
```

Comment:
is there no parameter, that need to be change dynamically ?

Reply:
The `MeshNavigationServer` only uses the parameters from the `AbstractNavigationServer`, so it does not define any additional parameters.
With ROS2, the reconfiguration of the basic params from `AbstractNavigationServer` are transparently handled there. I think this is one of the benefits from the ROS2 param scheme over how it needed to be done in ROS1 with the `dynamic_reconfigure` server. See the cpp code changes: https://github.com/naturerobots/mesh_navigation/pull/38/files#diff-157b1f4fed5d36211585e7c4dad7f28fa5962b98be303e5f5d65cba84495e355L232
Previously, we had this "to abstract" method that pipes the abstract parts of the parameters through to the `AbstractNavigationServer`. I think this approach was necessary because the `dynamic_reconfigure` server only supported a *single* callback. In ROS2, we can use any number of callbacks that only handle parts of the parameters.